src/ng/filter/filters.js

```diff
@@ -331,11 +331,15 @@ function formatNumber(number, pattern, groupSep, decimalSep, fractionSize) {
   }
 }
 
-function padNumber(num, digits, trim) {
+function padNumber(num, digits, trim, negWrap) {
   var neg = '';
   if (num < 0) {
-    neg =  '-';
-    num = -num;
+    if (negWrap) {
+      num = -num + 1;
```

Comment:
Is this `+1` necessary ? I think it's not consistent with how browsers treat negative years.